crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -747,8 +750,21 @@ pub fn bridge_config(
                     }
                 };
 
+            let use_mqtt_service = c8y_config.mqtt_service.enabled;
+            let mut mqtt_host = c8y_config.mqtt.or_config_not_set()?.clone();
+
+            // If the MQTT URL is still pointing to the default core MQTT port, when MQTT service is enabled
+            // implicitly switch to the MQTT service port.
+            // When the port is not the default one,
+            // we assume that the user has explicitly configured it to point to the MQTT service endpoint.
+            if use_mqtt_service && mqtt_host.port().0 == MQTT_CORE_TLS_PORT {
+                let mqtt_svc_host = format!("{}:{}", mqtt_host.host(), MQTT_SVC_TLS_PORT);
+                mqtt_host = HostPort::<MQTT_TLS_PORT>::try_from(mqtt_svc_host)
+                    .map_err(TEdgeConfigError::from)?;
```

Comment:
This feels like a very brute-force approach to changing the port, where we generate a completely new string, then parse it. Can we not add a method such as `HostPort::set_port` to handle this case more cleanly? It might also make the code obvious enough we don't need such a large comment explaining what we're doing here